publish/publish.proj

```diff
@@ -23,7 +23,7 @@
 
   <Target Name="PublishToAzure"
           Condition="'$(Finalize)' != 'true'"
-          DependsOnTargets="GatherPublishToAzureBinaries;GatherPublishToAzureInstallers">
+          DependsOnTargets="GatherPublishToAzureBinaries;GatherPublishToAzureInstallers;GatherPublishToAzureChecksums">
```

Comment:
I know this pattern is already existing, but I find it quiet odd and confusing that publish.proj depends on this target in dir.targets, which then calls back into publish.proj.  It just seems too circular and over-engineered for what needs to be done.

You don't need to fix it in this PR, but I just wanted to state my opinion on this pattern.